aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -340,6 +340,7 @@ static std::tuple<Tensor &,Tensor &> max_out_impl(Tensor& max, Tensor& max_indic
               max_indices.device(), " for indices output");
   dim = maybe_wrap_dim(dim, self.dim());
   if (_dimreduce_return_trivial_no_ident(max, self, dim, keepdim, "max")) {
+    TORCH_CHECK(!self.is_complex(), "max does not support complex inputs.");
```

Comment:
It's worth noting that dispatch handlers are invoked in the methods registered for `min_stub` & `max_stub`.
However, the tests fail in some conditional cases such as this one, in which these stubs are not being called!

We can't rely upon dispatch macros to raise exceptions if they're never invoked in the first place!
So, in these corner cases, we need dtype checks.

Reply:
Great, thanks for investigating!